hazelcast/src/main/java/com/hazelcast/config/DiscoveryConfig.java

```diff
@@ -72,6 +73,9 @@ public NodeFilter getNodeFilter() {
     }
 
     public DiscoveryConfig setNodeFilter(NodeFilter nodeFilter) {
+        if (nodeFilter != null) {
```

Comment:
This null checks looks confusing.
The `EvictionConfig` setComparatorClassName is lot simpler. When I look the other methods, I think it makes sense to apply `@Nonnull`  annotation and add `checkNotNull` methods to them. 

EvictionConfig.set methods are also missing `@Nonnull` . 

@alex-dukhno  @devozerov What do you think ?

Reply:
@sancar I like the idea and I have updated code accordingly. Thank you for review